src/main/kotlin/org/wycliffeassociates/sourceaudio/upload/RequestHandler.kt

```diff
@@ -0,0 +1,76 @@
+package org.wycliffeassociates.sourceaudio.upload
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import io.ktor.http.content.PartData
+import io.ktor.http.content.streamProvider
+import org.wycliffeassociates.sourceaudio.upload.model.FileUploadModel
+import java.io.File
+import java.lang.IllegalArgumentException
+
+class RequestHandler {
+
+    fun handleFormUpload(parts: List<PartData>): String {
+        val paramsMap: MutableMap<String, String> = mutableMapOf()
+        var file = File("./src/main/resources/uploads/")
+
+        parts.forEach { part ->
+            when (part) {
+                is PartData.FormItem -> {
+                    paramsMap[part.name!!] = part.value
+                }
+                is PartData.FileItem -> {
+                    file = file.resolve(part.originalFileName!!)
+                    part.streamProvider().use { input ->
+                        file.outputStream().buffered().use {
+                            input.copyTo(it)
+                        }
+                    }
+                }
+            }
+            part.dispose()
+        }
+
+        return processUploadModel(file, paramsMap)
+    }
+
+    private fun processUploadModel(file: File, params: Map<String, String>): String {
+        var response = mutableMapOf<String, Any>()
```

Comment:
I believe you can change this to be val instead of var. This is because when you edit the variable, you are actually changing the map object that the variable points to in memory. You don't actually edit the variable in any way (i.e. you don't change where it is pointing so response never actually changes, technically).

Reply:
Yeah I missed that one when working with mutablemap and stuff